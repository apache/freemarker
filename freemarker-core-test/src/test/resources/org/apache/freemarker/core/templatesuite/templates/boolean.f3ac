<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<html>
<head>
<title>FreeMarker: Boolean Values Test</title>
</head>
<body>

<p>A simple test follows:</p>

<p>${message}</p>

<#assign b=true>

<p>Now perform boolean tests:</p>

<p><#if b>
    b is true.<br />
<#else>
    b is false.<br />
</#if></p>

<p><#if false>
    This can't be!<br />
<#else>
    This makes sense.<br />
</#if></p>

<p><#if boolean3>
    boolean3 succeeded.<br />
<#else>
    boolean3 failed.<br />
</#if></p>

<p><#if boolean4>
    boolean4 succeeded.<br />
<#else>
    boolean4 failed.<br />
</#if></p>

<p><#if boolean4 || boolean5>
    boolean4 || boolean5 succeeded.<br />
<#else>
    boolean4 || boolean5 failed.<br />
</#if></p>

<p><#if boolean5 || boolean4 || boolean5>
    boolean5 || boolean4 || boolean5 succeeded.<br />
<#else>
    boolean5 || boolean4 || boolean5 failed.<br />
</#if></p>

<p><#if boolean4 && boolean5>
    boolean4 && boolean5 succeeded.<br />
<#else>
    boolean4 && boolean5 failed.<br />
</#if></p>
<@assert true && true />
<@assert !(false && true) />
<@assert true and true />
<@assert !(false and true) />
 
<p>Now test list models:</p>

<p><#if list1??>
    list1 succeeded.<br />
<#else>
    list1 failed.<br />
</#if></p>

<p><#if list2??>
    list2 succeeded.<br />
<#else>
    list2 failed.<br />
</#if></p>

<p>Test hash models:</p>

<p><#if hash1??>
    hash1 succeeded: ${hash1.temp}<br />
<#else>
    hash1 failed: ${hash1.temp}<br />
</#if></p>

<p><#if hash2??>
    hash2 succeeded.<br />
<#else>
    hash2 failed.<br />
</#if></p>

<p>Test not operator:</p>

<p><#if ! boolean1>
   Not boolean1 succeeded
<#else>
   Not boolean1 failed
</#if></p>

<p><#if !boolean1>
   Not boolean1 succeeded
<#else>
   Not boolean1 failed
</#if></p>

<p><#if ! boolean2>
   Not boolean2 succeeded
<#else>
   Not boolean2 failed
</#if></p>

<p><#if !boolean2>
   Not boolean2 succeeded
<#else>
   Not boolean2 failed
</#if></p>

<p><#if message == "Hello, world!">
   Message is "Hello, world!"
<#else>
   Message is something else (${message})
</#if></p>

<p><#if message != "Hello, world!">
   Message is not "Hello, world!" -- it's ${message}
<#else>
   Message is "Hello, world!"
</#if></p>

</body>
</html>
