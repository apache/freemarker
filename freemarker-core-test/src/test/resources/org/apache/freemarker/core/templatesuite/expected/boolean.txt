/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
<html>
<head>
<title>FreeMarker: Boolean Values Test</title>
</head>
<body>

<p>A simple test follows:</p>

<p>Hello, world!</p>


<p>Now perform boolean tests:</p>

<p>
    b is true.<br />
</p>

<p>    This makes sense.<br />
</p>

<p>
    boolean3 succeeded.<br />
</p>

<p>
    boolean4 succeeded.<br />
</p>

<p>
    boolean4 || boolean5 succeeded.<br />
</p>

<p>
    boolean5 || boolean4 || boolean5 succeeded.<br />
</p>

<p>    boolean4 && boolean5 failed.<br />
</p>
 
<p>Now test list models:</p>

<p>
    list1 succeeded.<br />
</p>

<p>
    list2 succeeded.<br />
</p>

<p>Test hash models:</p>

<p>
    hash1 succeeded: Hello, world.<br />
</p>

<p>
    hash2 succeeded.<br />
</p>

<p>Test not operator:</p>

<p>
   Not boolean1 succeeded
</p>

<p>
   Not boolean1 succeeded
</p>

<p>   Not boolean2 failed
</p>

<p>   Not boolean2 failed
</p>

<p>
   Message is "Hello, world!"
</p>

<p>   Message is "Hello, world!"
</p>

</body>
</html>
